app/models/customer.rb

```diff
@@ -3,4 +3,11 @@ class Customer < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  after_commit :assign_customer_id, on: :create
```

Comment:
Just FYI, from API Docs: "This callback is called after a record has been created, updated, or destroyed".
Here we're specifying it to run the `assign_customer_id` action on the creation of a Customer record.